//  File: cloud_connection.proto
//  Author: Kevin Walsh <kwalsh@holycross.edu>
//
//  Description: Protocol buffers for CloudChannel.
//
//  Copyright (c) 2014, Kevin Walsh.  All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package cloudproxy;

// Types of CloudChannelFrame messages.
enum CloudChannelFrameTag {
  // permission to perform any operation on any object
  CLOUD_CHANNEL_FRAME_HANDSHAKE = 1;
  CLOUD_CHANNEL_FRAME_WRAPPED = 2;
  CLOUD_CHANNEL_FRAME_ABORT = 3;
  CLOUD_CHANNEL_FRAME_SHUTDOWN = 4;
}

// A tagged frame of data that contains handshaking and other
// connection-management messages or a wrapped message for some higher layer..
message CloudChannelFrame  {
	required CloudChannelFrameTag tag = 1;
	required string msg = 2;
}
